python/semantic_kernel/planners/function_calling_stepwise_planner/function_calling_stepwise_planner.py

```diff
@@ -0,0 +1,249 @@
+    step_prompt: str
+
+    def __init__(self, service_id: str, options: Optional[FunctionCallingStepwisePlannerOptions] = None):
+        """Initialize a new instance of the FunctionCallingStepwisePlanner
+
+        If the options are configured to use callbacks to get the initial plan and the step prompt,
+        the planner will use those provided callbacks to get that information. Otherwise it will
+        read from the default yaml plan file and the step prompt file.
+
+        Args:
+            service_id (str): The service id
+            options (Optional[FunctionCallingStepwisePlannerOptions], optional): The options for
+                the function calling stepwise planner. Defaults to None.
+        """
+        options = options or FunctionCallingStepwisePlannerOptions()
+        generate_plan_yaml = (
+            options.get_initial_plan() if options.get_initial_plan else open(PLAN_YAML_FILE_PATH).read()
+        )
+        step_prompt = options.get_step_prompt() if options.get_step_prompt else open(STEP_PROMPT_FILE_PATH).read()
+        options.excluded_plugins.add(STEPWISE_PLANNER_PLUGIN_NAME)
+
+        super().__init__(
+            service_id=service_id,
+            options=options,
+            generate_plan_yaml=generate_plan_yaml,
+            step_prompt=step_prompt,
+        )
+
+    async def execute(
+        self,
+        kernel: Kernel,
+        question: str,
+    ) -> FunctionCallingStepwisePlannerResult:
+        """
+        Execute the function calling stepwise planner
+
+        Args:
+            kernel: The kernel instance
+            question: The input question
+
+        Returns:
+            FunctionCallingStepwisePlannerResult: The result of the function calling stepwise planner
+
+        Raises:
+            PlannerInvalidConfigurationError: If the input question is empty
+        """
+        if not question:
+            raise PlannerInvalidConfigurationError("Input question cannot be empty")
+
+        chat_completion = kernel.get_service(service_id=self.service_id)
+        prompt_execution_settings: OpenAIChatPromptExecutionSettings = (
+            self.options.execution_settings
+            or kernel.get_service(self.service_id).get_prompt_execution_settings_class()(service_id=self.service_id)
+        )
+
+        # Clone the kernel so that we can add planner-specific plugins without affecting the original kernel instance
+        cloned_kernel = copy(kernel)
+        cloned_kernel.import_plugin_from_object(UserInteraction(), "UserInteraction")
+
+        # Create and invoke a kernel function to generate the initial plan
+        initial_plan = await self._generate_plan(question, cloned_kernel)
+
+        chat_history_for_steps = await self._build_chat_history_for_step(question, initial_plan, cloned_kernel)
+        prompt_execution_settings.tool_choice = "auto"
+        prompt_execution_settings.tools = get_tool_call_object(kernel, {"exclude_plugin": [self.service_id]})
```

Comment:
is the service_id the same as the plugin_name for this one?

Reply:
yes